deps/rabbit_common/src/rabbit_net.erl

```diff
@@ -168,7 +168,11 @@ port_command(Sock, Data) when ?IS_SSL(Sock) ->
         {error, Reason} -> erlang:error(Reason)
     end;
 port_command(Sock, Data) when is_port(Sock) ->
-    erlang:port_command(Sock, Data).
+    case gen_tcp:send(Sock, Data) of
+        ok -> self() ! {inet_reply, Sock, ok},
+              true;
+        {error, Reason} -> erlang:error(Reason)
+    end.
```

Comment:
1. If we support OTP 25 as minimum in 3.12, shouldn't we keep the old behaviour for OTP 25 and apply the new behaviour for OTP 26 (e.g. by checking macro `?OTP_RELEASE`). Otherwise, we may end up with performance regressions running this code in OTP 25 which does not make use of selective receive optimisations.
2. If we decide to keep this change as-is, why do we send `{inet_reply, Sock, ok}` to our own process? Can we remove the handling of all `inet_reply` messages in the protocol readers and writers?
3. If we decide to keep this change as-is, what's the difference between `sync` and `async` in https://github.com/rabbitmq/rabbitmq-server/blob/0f8a5899de55b928f67d31cdd14ea6c025e3678d/deps/rabbitmq_stomp/src/rabbit_stomp_reader.erl#L374-L384 Can we simply this function?
4. Can we delete `rabbit_net:port_command/2` and just use `rabbit_net:send/2` instead?

Reply:
I pushed a commit which fixes the regressions mentioned in 1.
2.-4. will be fixed once we require OTP 26 as minimum.